cql3/restrictions/statement_restrictions.cc

```diff
@@ -1012,8 +1012,18 @@ dht::partition_range_vector partition_ranges_from_token(const expr::expression&
     return {{std::move(start), std::move(end)}};
 }
 
+void validate_key_length(const std::vector<managed_bytes>& key, const size_t max_length) {
+    for (auto k : key) {
+        if (k.size() > max_length) {
+            throw exceptions::invalid_request_exception(format("Key length of {} is longer than maximum of {}", k.size(), max_length));
+        }
+    }
+}
+
 /// Turns a partition-key value into a partition_range. \p pk must have elements for all partition columns.
 dht::partition_range range_from_bytes(const schema& schema, const std::vector<managed_bytes>& pk) {
+    // partition key length can't be  greater then 65535
```

Comment:
Nitpick: I think the comments every time you calll validate_key_length are unnecessary. The call is pretty self-explanatory, and if you think it isn't, you can add a comment in one place, before validate_key_length(), explaining what this function validates.